pkg/transport/filters/filter_factory.go

```diff
@@ -0,0 +1,71 @@
+// SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors.
+//
+// SPDX-License-Identifier: Apache-2.0
+package filters
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	// ComponentNameFilterType defines the type of a component name filter
+	ComponentNameFilterType = "ComponentNameFilter"
+
+	// ResourceTypeFilterType defines the type of a resource type filter
+	ResourceTypeFilterType = "ResourceTypeFilter"
+
+	// AccessTypeFilterType defines the type of a access type filter
+	AccessTypeFilterType = "AccessTypeFilter"
```

Comment:
when adding a new filter, here and l34f should be modified. maybe add a comment like place new filters here?